### PR TITLE
fix: handle error in label-transform when using empty data

### LINKED ISSUE #3064: Bug in vega-label for empty data
I would like to report a bug in vega-label.

**Description**:
When I apply a label transform on underlying marks, it works perfectly unless there is no underlying marks.

Here is a working example : no bug [here](https://vega.github.io/editor/#/url/vega/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykSArJQBWENgDsQAGhAATONABONHJnaT0AQQAETJBBpRdSRdtgnM2uAA8kCHABs4U7QHc6MbcSQOGcbQ5ITHAOENoQMGyu4toMOBLaCGwMEP6RpIqU0iDusvRoACwADEUy8DRkWGgATCUyOEiysjTiZGgCMrJImCioANqg4nZwaCA9TE7Z3r4KaAMgUN3kbIoAnqOa2XbJ4pg1ABwlAgC+UqCLmMtrowBCW0kMu+0CR6fnS2Qr6+gAwvc7e1QAFoCgBmEpFN4LD5fUYAEX+j0BAE4AIwQqEXK7fEAAUURT1Q+3R6MxMOu6AAYgTAQJwaSztDLp8KSAAOI0tBA1HIoq1MnM2HoAASnKJAHYirzjgBdKFDBAjdDjJzVKY+PwQOZymUyAxkIahbV6xZOLX9QbDUbWaA+EYyTCrHBKkB6cSybKyNjIFpoUBdHqjFX2kAAMxoIQ96CxLPWpxAiiQrRduXy9UazVaaCKlCKHQTOyjmEUfnlVvQq1tk063qQvtQ-u6vTGQWrYYjDijIG2SJA8fEhiVxb8MkTydG5Uqe1lMiQ1lmFpAKwjhNdbGw3uyVZdNtNIyhy7gq6cob2Jrto0re+yZGURZLcBnIGQigA1ub5o7naNFHAoGeQAVF09EULUZFDRRNwbOQmyDVt9xkI8oDYeQ-RAOIAyVUBrDQ7drTw8COy7GNYXjVMYFw690F3C8ZDdKNUXjb5QDwisCPbSNRh7J4mLVaDWJAK9aJAaY-GzeNwwcBw0NEl0vVkMg4EUCYxOONSoS-F1LmsACIKgxtA3QECtXjJCUKw9CcEwtCfAqDRQFk0YoCPS5FD7Oj9BCFoLMcoz10wTd420wEWLsnxRlDFZkEwAAKAMGAQSh4sS7jMCkAByShqggdKAEp3PbKSZI1F1fw9CSJEwABlGgAC8fJKtBUWqdSHTHCBIsUBA5lATTRkCYJpNnYg2BoWQAFkTHfOZXRMLVdW7cRYBWGaApwEAFrYUNQ1SQE+lRBaDHq3Cwuk9A+nI7QAGptAANlKbRJywGUCoaJoWjaVBSR1Y4gA)
Now, here is the same example but the barchart gets its data from an empty table : [issue here](https://vega.github.io/editor/#/url/vega/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykSArJQBWENgDsQAGhAATONABONHJnaT0AQQAETJBBpRdSRdtgnM2uAA8kCHABs4U7QHc6MbcSQOGcbQ5ITHAOENoQMGyu4toMOBLaCGwMEP6RpIqU0iDusvRoACwADEUy8DRkWGgATCUyOEiysjTiZGgCMrJImCioANqg4nZwaCA9TE7Z3r4KaAMgUN3kbIoAnqOa2XbJ4pg1ABwlAgC+UqCLmMtrowBCW0kMu+0CR6fnS2Qr6+gAwvc7e1QAFoCgBmEpFN4LD5fUYAEX+j0BAE4AIwQqEXK7fEAAUURT1Q+3R6MxMOu6AAYgTAQJwaSztDLp8KSAAOI0tBA1HIoq1MnM2HoAASnKJAHYirzjgBdKFDBAjdDjJzVKY+PwQOZymUyAxkIahbV6xZOLX9QbDUbWaA+EYyTCrHBKkB6cSybKyNjIFpoUBdHqjFX2kAAMxoIQ96CxLPWpxAiiQrRduXy9UazVaaCKlCKHQTOyjmEUfnlVvQq1tk063qQvtQ-u6vTGQWrYYjDijIG2SJA8fEhiVxb8MkTydG5Uqe1lMiQ1lmFpAKwjhNdbGw3uyVZdNtNIyhy7gq6cob2Jrto0re+yZGURZLcBnIGQigA1ub5o7naNFHAoGeQAVF09EULUZFDRRNwbOQmyDVs4DVeMjygNh5D9EA4gDJVQGsdDt2tfDwI7LsY1heNUxgPDr3QXcLxkN0o1ReNvlAfCK0I9tI1GHsnmYtVoLYkArzokBpj8bN43DBwHHQsSXS9WQyDgRQJnE451KhL8XUuawAIgqDG0DdAQK1JDxBQtDoMwpZ0J8CoNFAOTRigI9LkUPt6P0EIWmw0SNWA9dME3SSaGk2T-J-OAPUkiRMAAZRoAAvXynNQVFqg0h0xwgUMVgQOZQC00ZAmCGTZ2INgaFkABZEx3zmV0TC1XVu3MyJ3P6MY2BwEAWrYUNQ1SQE+lRFqDGSvD7J8UY+go7QAGptAANlKbRJywGUPJABomhaNo0shHVjiAA)

As you can see, we get the following error: 
[Error] Cannot read property 'length' of undefined

I believe this issue just need to protect code when the underlying marks list are empty and might be related to this line of code: [line 39 in LabelLayout](https://github.com/vega/vega/blob/ec61d7a21a5362e48eff875cba286a07f1a196ac/packages/vega-label/src/LabelLayout.js#L39)

### PR DESCRIPTION
**Pull request by chanwutk**
Fix #3064

### CODE REVIEW

File: packages/vega-label/src/Label.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ inherits(Label, Transform, {
 
     // run label layout
     labelLayout(
-      pulse.materialize(pulse.SOURCE).source,
+      pulse.materialize(pulse.SOURCE).source || [],
       _.size,
       _.sort,
       array(_.offset || 1),
```
Comment (domoritz) [inline packages/vega-label/src/Label.js:92]:
```suggestion
      array(_.offset ?? 1),
```

Otherwise, an offset of 0 does not work, right?

**Review by domoritz — APPROVED** [review]

File: packages/vega-label/src/Label.js (hunk 1 above)
Comment (chanwutk) [reply packages/vega-label/src/Label.js:92]: Yes, you are correct.

**Comment by chanwutk** [comment]
@domoritz It seems like ESlint cannot parse `??`

**Comment by domoritz** [comment]
Try with https://github.com/vega/vega/pull/3097

**Comment by chanwutk** [comment]
Should I wait for the #3097 to get merged first, or should I just rebase my branch to #3097 ?

**Comment by domoritz** [comment]
I think the two fixes are unrelated so maybe separate them into two prs.

File: packages/vega-label/src/Label.js [hunk 2]
```diff
@@ -86,13 +86,13 @@ inherits(Label, Transform, {
 
     // run label layout
     labelLayout(
-      pulse.materialize(pulse.SOURCE).source,
+      pulse.materialize(pulse.SOURCE).source || [],
       _.size,
       _.sort,
-      array(_.offset || 1),
+      array(_.offset === 0 ? 0 : (_.offset || 1)),
```
Comment (domoritz) [inline packages/vega-label/src/Label.js:92]:
```suggestion
      array(_.offset == null ? 1 : _.offset),
```

**Comment by domoritz** [comment]
Looks good to me

**Pull request merged**